website/content/docs/ecs/architecture.mdx

```diff
@@ -62,56 +76,103 @@ This diagram shows an example timeline of a task shutting down:
+* **Service tokens:** used by sidecar containers for service registration and health syncing
+
+With Consul on ECS, these tokens are obtained dynamically when a task starts up by logging
+in to Consul's AWS IAM auth method.
+
+### Consul Client Token
+
+Consul client tokens require `node:write` for any node name, which is necessary because the Consul node
+names on ECS are not known until runtime.
 
-### Automatic ACL Token Provisioning
+### Service Token
 
-Consul ACL tokens secure communication between agents and services.
-The following containers in a task require an ACL token:
+Service tokens are associated with a [service identity](/docs/security/acl#service-identities).
+The service identity includes `service:write` permissions for the service and sidecar proxy.
 
-- `consul-client`: The Consul client uses a token to authorize itself with Consul servers.
-  All `consul-client` containers share the same token.
-- `mesh-init`: The `mesh-init` container uses a token to register the service with Consul.
-  This token is unique for the Consul service, and is shared by instances of the service.
+### AWS IAM Auth Method
 
-The ACL controller automatically creates ACL tokens for mesh-enabled tasks in an ECS cluster.
-The `acl-controller` Terraform module creates the ACL controller task. The controller creates the
-ACL token used by `consul-client` containers at startup and then watches for tasks in the cluster. It checks tags
-to determine if the task is mesh-enabled. If so, it creates the service ACL token for the task, if the
-token does not yet exist.
+When a service mesh task on ECS starts up, it runs two `consul login` commands to obtain a client
+token and a service token. When the task stops, it attempts two `consul logout` commands in order
+to destroy these tokens.
 
-The ACL controller stores all ACL tokens in AWS Secrets Manager, and tasks are configured to pull these
-tokens from AWS Secrets Manager when they start.
+During a `consul login`, the [task's IAM
+role](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task-iam-roles.html) is presented
+to the AWS IAM auth method on the Consul servers. The role is validated with AWS. If the role is
+valid, and if the auth method trusts the IAM role, then the role is permitted to login. A new Consul
+ACL token is created and [Binding Rules](/docs/security/acl/auth-methods#binding-rules) associate
+permissions with the newly crated token. These permissions are mapped to the token based on the IAM
+role details. For example, tags on the IAM role are used to specify the service name and the
+namespace to be associated with a service token.
+
+When using the Consul on ECS Terraform modules, the auth method will be configured automatically
+by the ACL controller.
+
+## ACL Controller
+
+The ACL controller performs the following operations on the Consul servers:
+
+* Manages Consul admin partitions and namespaces.
```

Comment:
These are enterprise too

Reply:
I actually had two bullet points about ent stuff here, so cleaned that up